### DISCUSSION TITLE
Change order of asc and desc in grid sort

### DISCUSSION THREAD

**Discussion by Eralmidia** [IgniteUI/igniteui-angular#11903] [Q&A]
I was just wondering if there is any way to change the order of asc and desc in the grid is applied when clicking the sort arrow? Ideally on a application level.

For instance, we have grid a column which displays a failure count. The users of this grid is commonly interested only in the rows with the highest number of failures. But by default clicking on the sorting arrow in the column sorts ascending first, then descending on the second click, resulting in the users have to click twice every time they need to sort at grid.

First click results in only rows which are of now interest to the user:
![image](https://user-images.githubusercontent.com/12936286/180762672-9add176d-5fba-4f7f-b064-144c4e4d81f0.png)
 While second click gets us the interesting data:
 
![image](https://user-images.githubusercontent.com/12936286/180762944-70678d2f-7788-48fe-b548-32e8c1b0872c.png)

**Comment by ddaribo** [comment]
Hi @Eralmidia,
Thank you for reaching out!

Yes, this can be achieved on application level with a workaround approach. It involves [retemplating the sorting indicators](https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/sorting#sorting-indicators-templates) in order to easily add a `click` event handler to the icons. The default igx-icons can be specified, so that the original look is kept:

```
    <ng-template igxSortHeaderIcon let-header>
      <igx-icon (click)="handleCustomSortOrder(header)">arrow_upward</igx-icon>
    </ng-template>
    <ng-template igxSortAscendingHeaderIcon let-header>
      <igx-icon (click)="handleCustomSortOrder(header)">arrow_upward</igx-icon>
    </ng-template>
    <ng-template igxSortDescendingHeaderIcon let-header>
      <igx-icon (click)="handleCustomSortOrder(header)">arrow_downward</igx-icon>
    </ng-template>
```

The click handler:

```
public handleCustomSortOrder(header: IgxGridHeaderComponent) {
    const column = header.column;
    var nextdir = 1;
    if (column.sortable) {
      var sorted = this.grid.sortingExpressions.find(
        (x) => x.fieldName === column.field
      );

      if (sorted !== undefined) {
        if (sorted.dir === 1) {
          nextdir = 2;
        }

        if (sorted.dir === 2) {
          nextdir = 0;
        }

        this.grid.sort({
          fieldName: column.field,
          dir: nextdir,
          ignoreCase: true,
        });
      } else {
        this.grid.sort({
          fieldName: column.field,
          dir: nextdir,
          ignoreCase: true,
        });
      }
    }
  }
```

This essentially sorts by the next direction just before the grid performs the sorting, which results in seemingly “skipping” the default "next". Thus, the order becomes None -> Descending -> Ascending -> None.
However, please keep in mind that in case you are handling either the grid’s [sorting ](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/IgxGridComponent.html#sorting) or [sortingDone ](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/IgxGridComponent.html#sortingDone) events, they would be triggered twice for each sort operation, as two sorts are performed on each icon click.

[Here ](https://stackblitz.com/edit/igx-grid-sort-descending-first?file=src/app/grid/grid-sorting-indicators/grid-sorting-indicators.component.ts) you can find a small StackBlitz sample demonstrating this suggestion. I hope it helps!
